fabric-events-player-v1/src/main/java/net/fabricmc/mixin/event/player/MixinPlayerManager.java

```diff
@@ -0,0 +1,24 @@
+package net.fabricmc.mixin.event.player;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.dimension.DimensionType;
+
+import net.fabricmc.api.event.player.v1.PlayerRespawnCallback;
+
+@Mixin(PlayerManager.class)
+public abstract class MixinPlayerManager {
+	/**
+	 * The reason why this injection was chosen rather than injecting at the "TAIL" of copyFrom is to allow repositioning of the player before respawn.
+	 */
+	@Inject(at = @At("TAIL"), method = "respawnPlayer(Lnet/minecraft/server/network/ServerPlayerEntity;Lnet/minecraft/world/dimension/DimensionType;Z)Lnet/minecraft/server/network/ServerPlayerEntity;")
```

Comment:
On second thought, I believe the injection point could be improved. With `TAIL`, if a mod decides to reposition the player, the client will have to load the terrain all over and even show a loading screen if changing dimensions.
Moving the injection to right before the packets are sent ensures the game stays smooth even through heavy respawn alteration.
```suggestion
	@Inject(at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;getLevelProperties()Lnet/minecraft/world/level/LevelProperties;"), method = "respawnPlayer(Lnet/minecraft/server/network/ServerPlayerEntity;Lnet/minecraft/world/dimension/DimensionType;Z)Lnet/minecraft/server/network/ServerPlayerEntity;")
```
(should probably change the comment to reflect that too)

Reply:
Gave that a look and that is probably a better choice for injection, though I either have to use locals in this scenario or find the new player entity in the world

Reply:
True, that could complicate things a bit. Here is what I use, for reference : https://github.com/Ladysnake/Requiem/blob/master/src/main/java/ladysnake/requiem/mixin/server/PlayerManagerMixin.java#L164-L185
Finding the player in the world would have definitely made breakage less likely, however the player won't have been added by this point.